### PR TITLE
Debugging FAQ

### LINKED ISSUE #185: Add a heapster debugging FAQ
The current workflow is to debug heapster issues involves running heapster with extra logging. It needs to be documented to help users identify problems with their setup.

### PR DESCRIPTION
**Pull request by vishh**
This is currently a brain dump. Needs polishing.
Fixes #185

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Depending on the deployment setup, the issue could be either with heapster, cadvisor, kubernetes, or the monitoring backend.
+
+Ensure your monitoring backend is up and reachable. When running with the InfluxDB sink, heapster attempts to create a database by default, which will fail eventually after a fixed number of retries. 
+
+Heapster exports a '/validate' endpoint that will provide some information about its current state.
+
+If the '/validate' endpoint does not provide enough information, additional logging can be enabled by setting an extra flag. This requires restarting heapster though.
+Add `--vmodule=*=4` flag to heapster. When using the docker image or when running in kubernetes, pass an extra environment variable `FLAGS="--vmodule=*=4`. 
+If you are running heapster on kubernetes, the environment variable needs to be added to the `env` section in (heapster-controller.yaml)[../deploy/kube-config/heapster-controller.yaml]
```
Comment (vmarmol) [inline docs/debugging.md]: link the the other way around in Markdown

File: docs/debugging.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+## Heapster Debugging FAQ:
+
+This is a collection of common issues faced by users and ways to debug them.
+
+Depending on the deployment setup, the issue could be either with heapster, cadvisor, kubernetes, or the monitoring backend.
+
+Ensure your monitoring backend is up and reachable. When running with the InfluxDB sink, heapster attempts to create a database by default, which will fail eventually after a fixed number of retries.
```
Comment (vmarmol) [inline docs/debugging.md]: Can we separate into Heapster and Sinks(InfluxDB) sections? :)

File: docs/debugging.md (hunk 1 above)
Comment (vishh) [inline docs/debugging.md]: Ahh :) Time to go home I guess. Fixed.

File: docs/debugging.md (hunk 2 above)
Comment (vishh) [inline docs/debugging.md]: Good idea. Done.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**